### PR TITLE
Add command `gst-to-kore` to kevm_pyk

### PR DESCRIPTION
**Pull request by ehildenb**
A few things:

- Adds targets `venv`, `venv-dev`, and `venv-clean` as the "standard" targets across our Python repos.
- Factors out variable `$(PYK_ACTIVATE)` in the `Makefile`.
- Remove vestigial file `tests/gen-spec/foundry/foundry.toml` (we moved everything to directory `tests/foundry` instead).
- Renames argument parsers here to be consistent with upstream choice for `*_args`.
- Adds command `gst-to-kore` for consuming GeneralStateTests from the ethereum-tests submodule, and emitting the kore for execution. Uses factored out `evm_chain_args` arguments for setting schedule, mode, and chainid. This uses basically an exact copy of what `scripts/kore-json.py` is.
- Allows choosing to use `kevm_pyk gst-to-kore ...` instead of file `scripts/kore-json.py` for converting JSON test format to Kore.
- Adds test of `kevm kast --pyk ...` on CI.

This gives us a way to start measuring the performance of our kore data-structures. Currently, it's using string-building style approach (directly emitting to stdout), to stay fast. Some commands profiled (both with `--profile`, on without `--pyk` and one with):

```
% kevm interpret tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmSha3Test/sha3_bigOffset2.json --backend llvm                                            \ 
    --mode VMTESTS --schedule DEFAULT --chainid 1 --profile
0m0.018s 0m0.018s 0m0.000s kore-json.py tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmSha3Test/sha3_bigOffset2.json LblDEFAULT'Unds'EVM{}() LblVMTESTS{}() \dv{SortInt{}}("1")
0m0.007s 0m0.000s 0m0.007s /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm/interpreter /tmp/tmp.l2Poddp5Di -1 /tmp/tmp.bEIKwzJ7j6 --profile

% kevm interpret tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmSha3Test/sha3_bigOffset2.json --pyk --backend llvm                                            \
    --mode VMTESTS --schedule DEFAULT --chainid 1 --profile
0m0.061s 0m0.061s 0m0.000s python3 -m kevm_pyk gst-to-kore tests/ethereum-tests/LegacyTests/Constantinople/VMTests/vmSha3Test/sha3_bigOffset2.json --schedule DEFAULT --mode VMTESTS --chainid 1 --definition /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm -I /home/dev/src/evm-semantics/.build/usr/lib/kevm/include/kframework -I /home/dev/src/evm-semantics/.build/usr/lib/kevm/blockchain-k-plugin/include/kframework
0m0.008s 0m0.008s 0m0.000s /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm/interpreter /tmp/tmp.RW1sEUED8e -1 /tmp/tmp.CThZbJ9xSq --profile
```

The first command has no `--pyk`, and spends 18ms in `kore-json.py`. The second command has `--pyk` and spends 61ms in `kevm_pyk gst-to-kore`. So this introduces about 40ms overhead.

Across the entire test-suite, this is about a 1.48X slowdown to use `kevm_pyk gst-to-kore` instead of `kore-json.py`. This data was collected by using `make test-conformance -j20 KEVM='profile log kevm'` and `make test-conformance -j20 KEVM='profile log kevm' TEST_OPTIONS='--pyk'`, with `profile` executable defined here: https://github.com/runtimeverification/evm-semantics/issues/1165

### CODE REVIEW

File: kevm_pyk/src/kevm_pyk/gst_to_kore.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+def mode_to_kore(mode: str) -> str:
+    return f'Lbl{mode}' + "{}()"
+
+
+def gst_to_kore(gst_file: Path, out_stream: TextIOWrapper, schedule: str, mode: str, chainid: int) -> None:
+
+    with open(gst_file) as data_file:
+        data = json.load(data_file, object_pairs_hook=OrderedDict)
+
+    def escape(_data):
```
Comment (nishantjr) [inline kevm_pyk/src/kevm_pyk/gst_to_kore.py:28]: Please add typing for nested functions too. Should these begin with an `_`?

**Review by nishantjr — APPROVED** [review]
LGTM. Some additional typing requested.

File: kevm_pyk/src/kevm_pyk/gst_to_kore.py (hunk 1 above)
Comment (ehildenb) [reply kevm_pyk/src/kevm_pyk/gst_to_kore.py]: Done.

File: kevm_pyk/src/kevm_pyk/gst_to_kore.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    def _print_k_config_var(_data: str) -> None:
+        out_stream.write('\\dv{SortKConfigVar{}}("$' + _data + '")')
+
+    def _print_sort_injection(s1: str, s2: str, _data: Any, printer: Callable[[Any], None]) -> None:
+        out_stream.write('inj{Sort' + s1 + '{}, ' + 'Sort' + s2 + '{}}(')
+        printer(_data)
+        out_stream.write(')')
+
+    def _print_kast(_data: Any, sort='JSON'):
```
Comment (nishantjr) [inline kevm_pyk/src/kevm_pyk/gst_to_kore.py:46]:
```suggestion
    def _print_kast(_data: Any, sort='JSON') -> None:
```

**Review by nishantjr — APPROVED** [review]

**Pull request merged**